### DISCUSSION TITLE
I cannot connect Tasmota to an online Mqtt Broker

### DISCUSSION THREAD

**Discussion by Grupo-Atomos** [arendst/Tasmota#13180] [Q&A] [answered]
I cannot connect Tasmota to an online Mqtt Broker.
I've tried others to check if it was Broker Mqtt, but I realized that it wasn't.
When I use nodemcu with any code other than Tasmota, it works.
In some previous version I had succeeded but I can't in current versions.
The Broker I'm referring to is hivemq whose MQTT connection settings are:
Host: broker.mqttdashboard.com
TCP port: 1883
Websocket port: 8000

When I try this error appears:
Connection failed to http://www.mqtt-dashboard.com/:1883, rc -2.
What could it be?

**Comment by barbudor** [comment, marked as answer]
Why is there Http in the log message? 
You must specify the IP address or host name of the MQTT broker without anything else.
MQTT is using MQTT protocol not Http. Mqtt over web sockets are not supported

**Reply by barbudor** [reply]
I've just tested with `mqtthost broker.hivemq.com` as specified on the front page and it works

![image](https://user-images.githubusercontent.com/2996042/133963265-06b78222-1f15-4072-85f4-3b3335cc0963.png)

**Reply by barbudor** [reply]
It works fine also with `broker.mqtt-dashboard.com`

**Reply by barbudor** [reply]
Don't forget that :
> Testing and usage is for free but please do not use it for sensitive information because **everybody is allowed to subscribe to every topic**, including wildcard. 

Which include anyone posting `cmnd` commands to YOUR Tasmota !!!

If you want a free to use online broker, go and check Flepsi or MyQTThub. My personnal preference is for Flepsi.
Each account is segregated from other, nobody can't see your devices.
Note that those requires TLS encrypted connections which make it safer for online service.

**Reply by barbudor** [reply]
I just noticed that if you register with HiveMQ, you can have a free private broker for up to 100 devices. It also requires TLS encrypted connection.

**Reply by Grupo-Atomos** [reply]
Hello Beard, all in peace?
HTTP was just the multiple ways I've been trying to succeed in connecting to the broker. I had already tried the other way you recommended.
I'm here with Tasmota connected to the Wifi network with everything working and with the MQTT settings as you instructed, but it doesn't work!
 
These are my device settings:

![Config do tasmota fw](https://user-images.githubusercontent.com/13262479/134088681-79fbbc36-a078-4f6a-bc94-b70b501325b6.png)

Those are the answers I get:

![Resposta Tasmota fw](https://user-images.githubusercontent.com/13262479/134089228-37867d50-a4fd-406a-97e7-2570142659ac.png)

Obs.: The HiveMQ, free private, I already have and the intention to use it.
Thank you very much for now!

**Reply by ascillato** [reply]
Your device doesn't have the time in sync. Your device is able to connect to the internet? or you have your network blocked?

**Reply by Grupo-Atomos** [reply]
ascillato, I noticed this now!
00:00:38,192 RSL: RESULT = {"Time":"1970-01-01T00:00:38"}
I live in sao paulo Brazil.
I downloaded the firmware from http://ota.tasmota.com/tasmota/release/tasmota-BR.bin.
How do I fix this?

**Reply by ascillato** [reply]
You need to allow your Tasmota to connect to internet. If you haven't blocked your devices to connect to internet, that means that your router´s DHCP is giving wrong NTP settings to your devices. In that case, you can try to make your router to provide correct NTP settings. If that is not possible, you can set in Tasmota to use a fixed IP, so Tasmota will fallback to its predefined the NTP servers. Use commands IPADDRESS1 to IPADDRESS4 ( Please, check https://tasmota.github.io/docs/Commands/ )

**Reply by ascillato** [reply]
Also, you can try using the latest version from the development branch that has a lot of fixes too:  http://ota.tasmota.com/tasmota/tasmota-BR.bin

You can do an OTAurl to that.

**Reply by barbudor** [reply]
@ascillato Tasmota do not receive anymore the NTP settings from DHCP. This has been disabled quite some time ago because of so many badly configured routers.
However if the Tasmota device has Internet access, the build-in public NTP servers should work

@JurandirHora Please provide the result of the following commands:
```
NtpServer
WebSend [1.1.1.1] /
WebSend [www.google.com] /
```
The 1st one will return the list of configured NTP server in your Tasmota
The 2nd one will test if you have Internet access by sending a HTTP request to 1.1.1.1, should answer "Done"
The 3rd one will test if you have a DNS that resolve www.google.com to an IP address, should answer "Done"

**Reply by Grupo-Atomos** [reply]
The answer that are provided

![Resposta Tasmota 1 fw](https://user-images.githubusercontent.com/13262479/134156073-a35d8e47-d7c3-4cc8-bc1f-4e33494c0227.png)

**Reply by barbudor** [reply]
So port 80 works (WebSend) but you have firewall rules that prevent your Tasmota devcie to access NTP servers and MQTT servers
The problem is not in Tasmota but in your IT system

**Reply by barbudor** [reply]
But you have still entered the MQTT host wrongly
It should be `broker.mqtt-dashboard.com` or `broker.mqttdashboard.com` and only that. No `http`, no `/`, no `:`

**Reply by barbudor** [reply]
![image](https://user-images.githubusercontent.com/2996042/134157407-fda723ba-bec9-46e2-9a25-12c4330940e4.png)
![image](https://user-images.githubusercontent.com/2996042/134157537-3b768234-e879-4887-9d20-93f5508a962f.png)

**Reply by ascillato** [reply]
@barbudor 
> @ascillato Tasmota do not receive anymore the NTP settings from DHCP. This has been disabled quite some time ago because of so many badly configured routers.

Exactly, but @JurandirHora has Tasmota 9.5.0 and the NTP relaxing was implemented in 9.5.0.5 ( https://github.com/arendst/Tasmota/blob/development/CHANGELOG.md#changed-3 ). That is why, in https://github.com/arendst/Tasmota/discussions/13180#discussioncomment-1360143 I'm asking the user to update by OTAurl to latest Tasmota from the development branch.

@JurandirHora Please, update to latest and fix the mqtt broker address.

**Reply by Grupo-Atomos** [reply]
Good night gentlemen!
I updated Tasmota with the version recommended by @ascillato but the defect perciste.
see:

![Tasmota atualizado fw](https://user-images.githubusercontent.com/13262479/134287962-57772279-7687-43c6-b5d5-6580154b4407.png)

regarding firewall rules that could be preventing my Tasmota device from accessing NTP servers and MQTT servers, quoted by @barbudor; I built a simple code with the PubSubClient.h library and installed it on NODEMCU which communicated with the broker perfectly. If this thesis were correct, a code as elementary as this one would also not be able to communicate!


https://user-images.githubusercontent.com/13262479/134288023-f38bde5c-badf-43f8-b27b-86dbccc891b6.mp4

I thank you so much for helping me!

**Reply by barbudor** [reply]
Your Arduino sketch is using 'broker.mqttdashboard.com' but your Tasmota logs show 'mqtt-dashboard.com'
You still haven't configured Tasmota correctly like on my screen shot.

**Reply by Grupo-Atomos** [reply]
@barbudor, I did it according to your recommendation!
Response:
![Confi fw](https://user-images.githubusercontent.com/13262479/134383046-f3113542-5f54-4cf9-93b7-247018a61f83.png)

![Confi2 fw](https://user-images.githubusercontent.com/13262479/134383129-6bba5188-83b5-4a6e-b2a7-5c3001bf78ce.png)

**Reply by barbudor** [reply]
OK.
I have no idea what is wrong on your side.
Given that your Tasmota can't neither get NTP nor MQTT, but can make web request.
I noticed that Tasmota use 192.168.2.185 but the Arduino sketch is using 192.162.2.186
I have no more idea. Sorry

**Reply by barbudor** [reply]
broker.mqttdashboard.com does not requires a password.
Do you have one? If yes try to remove it with `mqttpassword "`

**Reply by Grupo-Atomos** [reply]
I had already removed the password and username.
if you have any groups that i can get help from i appreciate it!
if you want to have access to my pc through teamviewer to check some configuration, just ask.
for now thanks!

**Comment by Grupo-Atomos** [comment]
Good morning!
Regarding this open ticket, I want to add that the problem was only solved with the v9.3.1 version present on the website https://tasmota.github.io/docs/Download/#official-ota-server.
Just install, put the broker's address online and everything was working.
It's a pity that we don't know for sure what happened that my nodemcus couldn't communicate with an online mqtt broker.

**Reply by rj10rd** [reply]
Perhaps your host wrong
You can access the broker at:

Broker: broker.hivemq.com
TCP Port: 1883

And please tell what type of hivemq using
Public or HIVmq cloud personal?

**Comment by Grupo-Atomos** [comment]
Public!
I had to roll back the installation to tasmota 8.5 for it to work normally

**Reply by cmatiasmartinez** [reply]
Buen dia, tengo la ultima version de tasmota, y los brokers myqtthub no funciona, entre con el mqtt explorer y sin problemas se conecta, es decir es una falla del tamosta, soy nuevo en este mundo IOT, y es una pesima experiencia, voy a instalar la version 8.5 de tasmota como recomiendan para comprobar el funcionamiento.

**Reply by barbudor** [reply]
This discussion forum is English only
Please translate your question

**Reply by cmatiasmartinez** [reply]
Good morning, I have the latest version of tasmota, and the myqtthub brokers do not work, I entered with the mqtt explorer and it connects without problems, that is, it is a tamosta failure, I am new to this IOT world, and it is a terrible experience, I'm going to install version 8.5 of tasmota as recommended to check the operation

**Reply by barbudor** [reply]
@cmatiasmartinez don't follow this advice. It outdated and wrong. In the meantime the author has succeeded in using a cloud broker with recent tasmota version but didn't updated this thread
Please provide details
- which esp?  Which tasmota binary?  You need support for TLS which is not available by default in Tasmota for esp8266
I used myqtthub in the past successfully

**Reply by cmatiasmartinez** [reply]
@barbudor How do I get TLS support for Tasmota ESP8266?

**Reply by barbudor** [reply]
By compiling your own binary following the docs
https://tasmota.github.io/docs/TLS/
https://tasmota.github.io/docs/Compile-your-build/

**Reply by barbudor** [reply]
I Just checked that MyQttHub supports unevcrypted MQTT over standard port 1883 so you may want to try that before going to TLS

**Reply by cmatiasmartinez** [reply]
![Captura](https://user-images.githubusercontent.com/110979401/185639170-3fdff0fb-49c0-4a2a-898c-e71e060bf874.JPG)
I'm sending you an image like the tasmota mqtt configuration is complete, however it doesn't connect, I'm making some mistake.

**Reply by barbudor** [reply]
Works fine for me

```
00:00:00.001 HDW: ESP8266EX
00:00:00.042 CFG: Loaded from flash at F6, Count 8
00:00:00.053 Project tasmota - dev-4119 Version 12.1.0.1(bhome)-2_7_4_9(2022-08-19T00:01:30)
00:00:03.497 WIF: Connecting to AP1 BNETWLAN Channel 1 BSSId 88:C3:97:F2:25:EC in mode 11n as dev-4119...
00:00:04.753 WIF: Connected
00:00:05.004 HTP: Web server active on dev-4119 with IP address 192.168.168.199
17:34:40.002 MQT: Attempting connection...
17:34:40.161 MQT: Connected
17:34:40.164 MQT: tele/nodemcu/LWT = Online (retained)
17:34:40.166 MQT: cmnd/nodemcu/POWER = 
17:34:40.172 MQT: tele/nodemcu/INFO1 = {"Info1":{"Module":"Generic","Version":"12.1.0.1(bhome)","FallbackTopic":"cmnd/18FE34A6D683_fb/","GroupTopic":"cmnd/alldevices/"}}
17:34:40.179 MQT: tele/nodemcu/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"dev-4119","IPAddress":"192.168.168.199"}}
17:34:40.190 MQT: tele/nodemcu/INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":4}}
17:34:44.455 MQT: tele/nodemcu/STATE = {"Time":"2022-08-19T17:34:44+02:00","Uptime":"0T00:00:12","UptimeSec":12,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"BNETWLAN","BSSId":"88:C3:97:F2:25:EC","Channel":1,"Mode":"11n","RSSI":78,"Signal":-61,"LinkCount":1,"Downtime":"0T00:00:06"}}
17:34:58.118 MQT: stat/nodemcu/STATUS0 = {"Status":{"Module":0,"DeviceName":"dev-4119","FriendlyName":["dev-4119"],"Topic":"nodemcu","ButtonTopic":"0","Power":0,"PowerOnState":0,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":0,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":1,"PowerRetain":1,"InfoRetain":0,"StateRetain":0},"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"alldevices","OtaUrl":"http://192.168.168.244:8000/tasmota/prod/tasmota-bhome.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:00:26","StartupUTC":"2022-08-19T15:34:32","Sleep":50,"CfgHolder":4617,"BootCount":4,"BCResetTime":"2021-09-05T18:50:21","SaveCount":9,"SaveAddress":"F5000"},"StatusFWR":{"Version":"12.1.0.1(bhome)","BuildDateTime":"2022-08-19T00:01:30","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"451/699"},"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"192.168.168.244","LogPort":5140,"SSId":["BNETWLAN",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008238","2805C80001000600003C5A18002800000000","000080C4","00006000","00004000","00000000"]},"StatusMEM":{"ProgramSize":616,"Free":384,"Heap":26,"ProgramFlashSize":1024,"FlashSize":4096,"FlashChipId":"1640EF","FlashFrequency":40,"FlashMode":3,"Features":["00000809","8F8AC587","04268001","000000CF","000013C0","C000F981","00004084","00001000","54000020","00000000"],"Drivers":"1,2,3,4,5,6,8,9,10,12,16,18,19,22,24,26,27,29,30,35,37,38,45,62","Sensors":"1,2,3,4,5,6"},"StatusNET":{"Hostname":"dev-4119","IPAddress":"192.168.168.199","Gateway":"192.168.168.254","Subnetmask":"255.255.255.0","DNSServer1":"192.168.168.244","DNSServer2":"1.1.1.1","Mac":"3C:71:BF:2A:D0:17","Webserver":2,"HTTP_API":1,"WifiConfig":5,"WifiPower":17.0},"StatusMQT":{"MqttHost":"node02.myqtthub.com","MqttPort":1883,"MqttClientMask":"18FE34A6D683","MqttClient":"18FE34A6D683","MqttUser":"nodemcu","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4},"StatusTIM":{"UTC":"2022-08-19T15:34:58","Local":"2022-08-19T17:34:58","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":99,"Sunrise":"06:48","Sunset":"20:58"},"StatusSNS":{"Time":"2022-08-19T17:34:58+02:00"},"StatusSTS":{"Time":"2022-08-19T17:34:58+02:00","Uptime":"0T00:00:26","UptimeSec":26,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"BNETWLAN","BSSId":"88:C3:97:F2:25:EC","Channel":1,"Mode":"11n","RSSI":78,"Signal":-61,"LinkCount":1,"Downtime":"0T00:00:06"}}}
```
The `Status0` command was send from MyQttHub web GUI using "Publish to device"

Please send you lorg. 
What error message do you get in the console ?

![image](https://user-images.githubusercontent.com/2996042/185654826-5106f464-74e7-461c-bc6f-8aa7db1e8e45.png)

**Reply by cmatiasmartinez** [reply]
using the MqttWifiTimeout command from the Tasmota console we must put MqttWifiTimeout 5000 and solved.

Since version 9.1.0.2 changed MQTT Wifi connection timeout from 5000ms to 200ms
